### ISSUE TITLE
Got on_load_function_failed when trying to start release.

### ISSUE THREAD

**Issue by gremly** [riverrun/argon2_elixir#5]
Hi,

Actually I am trying to deploy my app using `:distillery`, when i generate my `build` i try to test my new release using `rel/myapp/bin/myapp console` and everything goes ok, but when i move that release from my builds machine to the production box it does not start and throws me this error:

```
2016-11-30 16:09:13 crash_report
    initial_call: {supervisor,kernel,['Argument__1']}
    pid: <0.1181.0>
    registered_name: []
    error_info: {exit,{on_load_function_failed,'Elixir.Argon2.Base'},[{gen_server,init_it,6,[{file,"gen_server.erl"},{line,352}]},{proc_lib,init_p_do_apply,3,[{file,"proc_lib.erl"},{line,247
}]}]}
    ancestors: [kernel_sup,<0.1156.0>]
    messages: []
    links: [<0.1157.0>]
    dictionary: []
    trap_exit: true
    status: running
    heap_size: 376
    stack_size: 27
    reductions: 117
2016-11-30 16:09:13 supervisor_report
    supervisor: {local,kernel_sup}
    errorContext: start_error
    reason: {on_load_function_failed,'Elixir.Argon2.Base'}
    offender: [{pid,undefined},{id,kernel_safe_sup},{mfargs,{supervisor,start_link,[{local,kernel_safe_sup},kernel,safe]}},{restart_type,permanent},{shutdown,infinity},{child_type,supervisor
}]
```

I though that was a problem related with `argon2_nif.so` library but that is present within `myapp/lib/argon2_elixir-0.11.0/priv` directory.

Using `ldd` i got this:
```sh
$ ldd myapp/lib/argon2_elixir-0.11.0/priv/argon2_nif.so
myapp/lib/argon2_elixir-0.11.0/priv/argon2_nif.so:
	libthr.so.3 => /lib/libthr.so.3 (0x801209000)
	libc.so.7 => /lib/libc.so.7 (0x800823000)
```
And those libraries are present in that box, so i am so confused about what could be wrong :(.

May you have some thoughts about it?

Actually I am working with `FreeBSD 11.0-RELEASE-p1 (GENERIC)  amd64` with `Erlang/OTP 19 [erts-8.1] [source] [64-bit]` and `Elixir (1.3.4)`.

Thank you in advance.

**Comment by riverrun** [comment]
Are you building the so file on the production machine?

**Comment by gremly** [comment]
No, I don't. I use a build machine with the same architecture and operative system where i have installed Erlang/OTP and Elixir. The production machine has a base FreeBSD setup.

**Comment by riverrun** [comment]
Could you try building on the production machine and see what happens?
There might be small discrepancies in the setup.

**Comment by gremly** [comment]
Sure, i can build and start the release from the production machine. But i had to install all dependences a mean Erlang/Elixir, GCC, etc. My approach is just to generate a release which contains everything required by my app and then just unpack the tarball in production.

**Comment by riverrun** [comment]
Ok. It looks like there's some minor difference between the setup on the build machine and the production machine.

You said that you're using Erlang 19. Are you using the same minor version?

**Comment by riverrun** [comment]
I need to find out more about how distillery works. I'll get back to you next week. Sorry it can't be sooner -- the day job is keeping me really busy :(

**Comment by riverrun** [comment]
I should have asked this earlier, but can you deploy comeonin using the same setup?

**Comment by gremly** [comment]
Yes, I tried comeonin using the same setup and it works like a charm.

**Comment by riverrun** [comment]
Just to confirm: does the Erlang version that you're using support dirty schedulers?

Could you capture the `gcc ...` line when argon2_elixir is being built and let me know what it is? I just want to make sure that the C compiler is being called correctly.

**Comment by gremly** [comment]
Finally works in my production machine.

You were right, it was a problem related to dirty schedulers. I tried to compile it in my production machine, enabling dirty schedulers, but it seems that it wasn't enabled at all.

```
$ erl
Erlang/OTP 19 [erts-8.1] [source] [64-bit] [async-threads:10] [hipe] [kernel-poll:false]
```

It was in a single core instance, then I tried to change my instance type to a bigger one (2 cores) and it works like a charm. Although I saw a related problem in this [thread](http://erlang.org/pipermail/erlang-bugs/2015-December/005129.html).

Thanks for your help.
